### PR TITLE
Docs - provide section on tracking dependency failures in preview & published docs

### LINKED ISSUE #116: Provide dependency tracking failure
**Is your feature request related to a problem? Please describe.**
Currently, we have made it quite easier to track dependency durations with the `DependencyMeasurement` component. But this only is the case when the dependency doesn't fail during the tracking.

Example:
```csharp
using (var measurement = DependencyMeasurement.Start())
{
    // Do something on your dependency...
    // Possible exception throwal...

    _logger.LogDependency("SendGrid", "https://my.sendgrid.url", isSuccessful: true, startTime: measurement.StartTime, duration: measurement.Elapsed);
}
```
Onlly when the dependency interaction was sucessful will the tracking happen.

**Describe the solution you'd like**
We should make use of the `isSuccessful` parameter in some way so we could track failuers automatically.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Closes #116

### CODE REVIEW

File: docs/preview/features/writing-different-telemetry-types.md [hunk 1]
```diff
@@ -257,11 +257,31 @@ using (var measurement = DependencyMeasurement.Start())
...
-    object dependencyData = "http://my.sendgrid.uri/"
+    object dependencyData = "https://my.sendgrid.uri/";
     _logger.LogDependency("SendGrid", dependencyData, isSuccessful: true, startTime: measurement, context: telemetryContext);
 }
 ```
 
+Failures during the interaction with the tracked dependency can be controlled by the passed-allong `boolean`:
+
+```csharp
+string dependencyName = "SendGrid";
```
Comment (tomkerkhove) [inline docs/preview/features/writing-different-telemetry-types.md:268]: This is not used here

File: docs/features/writing-different-telemetry-types.md [hunk 2]
```diff
@@ -257,11 +257,31 @@ using (var measurement = DependencyMeasurement.Start())
...
-    object dependencyData = "http://my.sendgrid.uri/"
+    object dependencyData = "https://my.sendgrid.uri/"
     _logger.LogDependency("SendGrid", dependencyData, isSuccessful: true, startTime: measurement, context: telemetryContext);
 }
 ```
 
+Failures during the interaction with the tracked dependency can be controlled by the passed-allong `boolean`:
+
+```csharp
+string dependencyName = "SendGrid";
```
Comment (tomkerkhove) [inline docs/features/writing-different-telemetry-types.md:268]: This is not used here

File: docs/features/writing-different-telemetry-types.md [hunk 3]
```diff
@@ -257,11 +257,31 @@ using (var measurement = DependencyMeasurement.Start())
 
     // Track dependency
     string dependencyName = "SendGrid";
-    object dependencyData = "http://my.sendgrid.uri/"
+    object dependencyData = "https://my.sendgrid.uri/"
     _logger.LogDependency("SendGrid", dependencyData, isSuccessful: true, startTime: measurement, context: telemetryContext);
 }
 ```
 
+Failures during the interaction with the tracked dependency can be controlled by the passed-allong `boolean`:
```
Comment (tomkerkhove) [inline docs/features/writing-different-telemetry-types.md:265]:
```suggestion
Failures during the interaction with the tracked dependency can be controlled by passing `isSuccessful`:
```

File: docs/preview/features/writing-different-telemetry-types.md [hunk 4]
```diff
@@ -257,11 +257,31 @@ using (var measurement = DependencyMeasurement.Start())
 
     // Track dependency
     string dependencyName = "SendGrid";
-    object dependencyData = "http://my.sendgrid.uri/"
+    object dependencyData = "https://my.sendgrid.uri/";
     _logger.LogDependency("SendGrid", dependencyData, isSuccessful: true, startTime: measurement, context: telemetryContext);
 }
 ```
 
+Failures during the interaction with the tracked dependency can be controlled by the passed-allong `boolean`:
```
Comment (tomkerkhove) [inline docs/preview/features/writing-different-telemetry-types.md:265]:
```suggestion
Failures during the interaction with the tracked dependency can be controlled by passing `isSuccessful`:
```

File: docs/features/writing-different-telemetry-types.md (hunk 2 above)
Comment (stijnmoreels) [reply docs/features/writing-different-telemetry-types.md:268]: Yes, thanks. Updated that.

File: docs/preview/features/writing-different-telemetry-types.md (hunk 1 above)
Comment (stijnmoreels) [reply docs/preview/features/writing-different-telemetry-types.md:268]: Yes, thanks. Updated that.

**Pull request merged**